test/deploy.py

```diff
@@ -0,0 +1,486 @@
+        return vm_name #vm_name to refer to the newly create vm.
+
+    def checkCallOnVm(self, vm_name:string, cmd:list, failPrint=True) -> tuple:
+        '''Execute Command on VM.'''
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        if not self.client.instances.exists(vm_name):
+            raise AssertionError("VM {} does not exist.".format(vm_name))
+
+        inner_cmd = [
+            'lxc', 'exec', vm_name, '--', *cmd
+        ]
+        try:
+            subprocess.check_call(inner_cmd)
+        except subprocess.CalledProcessError as e:
+            if failPrint:
+                print("Failed Executing on {}: Output {}".format(vm_name, e))
+            raise e
+
+    def checkOutputOnCephadmShell(
+        self, vm_name:string, cmd:list, failPrint=True
+    ) -> dict:
+        '''Execute cmd on cephadm and return output'''
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        if not self.client.instances.exists(vm_name):
+            raise AssertionError("VM {} does not exist.".format(vm_name))
+
+        inner_cmd = [
+            'lxc', 'exec', vm_name, '--', 'cephadm', 'shell', *cmd
+        ]
+        try:
+            return subprocess.check_output(inner_cmd).decode('UTF-8')
+        except subprocess.CalledProcessError as e:
+            if failPrint:
+                print("Failed Cephadm Execution on {}: Output {}".format(vm_name, e))
+            raise e
+
+    def waitVmReady(self, vm_name) -> None:
+        isVmReady = False
+        counter = 0
+        while(not isVmReady):
+            try:
+                self.checkCallOnVm(vm_name, ['ls'], failPrint=False)
+                isVmReady = True
+            except subprocess.CalledProcessError as e:
+                counter+=1
+                print("Attempt {}: VM not ready".format(counter))
+                time.sleep(5) # Sleep for 5 sec.
+
+    def pushToVmRecursively(
+        self, vm_name:string, src_path:string, target_path:string
+    ) -> None:
+        '''Send Files (recursively) to VM'''
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        if not self.client.instances.exists(vm_name):
+            raise AssertionError("VM {} does not exist.".format(vm_name))
+
+        cmd = [
+            'lxc', 'file', 'push', src_path,
+            '{}{}'.format(vm_name, target_path), '-r'
+        ]
+        try:
+            subprocess.check_call(cmd)
+        except subprocess.CalledProcessError as e:
+            print("Failed Pushing {} to {}: Output {}".format(src_path, vm_name, e))
+            raise e
+
+    def modifyUserPasswd(
+        self, vm_name, user_name='ubuntu', new_passwd='ubuntu'
+    ) -> None:
+        '''Change user password for ssh access to LXD VM. '''
+        cmd = [
+            'echo', '{}:{}'.format(user_name, new_passwd), '| ' 'chpasswd'
```

Comment:
You're missing a comma I believe 
Also, piping with check_call only will work if you pass it as a shell string to execute, something in the form of:

```
check_call("echo foo:bar | chpasswd", shell=True)
```

Reply:
This method can be removed since it was added to enable pswd based ssh'ing into the vm.